src/main/java/org/apache/maven/plugins/dependency/ListClassesMojo.java

```diff
@@ -0,0 +1,402 @@
+
+    /**
+     * Repositories in the format id::[layout]::url or just URLs, separated by comma. ie.
+     * central::default::https://repo.maven.apache.org/maven2,myrepo::::https://repo.acme.com,https://repo.acme2.com
+     */
+    @Parameter( property = "remoteRepositories" )
+    private String remoteRepositories;
+
+    /**
+     * A string of the form groupId:artifactId:version[:packaging[:classifier]].
+     */
+    @Parameter( property = "artifact" )
+    private String artifact;
+
+    @Parameter( defaultValue = "${project.remoteArtifactRepositories}", readonly = true, required = true )
+    private List<ArtifactRepository> pomRemoteRepositories;
+
+    /**
+     * Download transitively, retrieving the specified artifact and all of its dependencies.
+     */
+    @Parameter( property = "transitive", defaultValue = "false" )
+    private boolean transitive = false;
+
+    /**
+     * Skip plugin execution completely.
+     *
+     */
+    @Parameter( property = "mdep.skip", defaultValue = "false" )
+    private boolean skip;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException
+    {
+        ProjectBuildingRequest buildingRequest = buildBuildingRequest();
+        DefaultDependableCoordinate coordinate = getCoordinate();
+
+        try
+        {
+            if ( isTransitive() )
+            {
+                Iterable<ArtifactResult> artifacts = getDependencyResolver()
+                        .resolveDependencies( buildingRequest, coordinate, null );
+
+                for ( ArtifactResult result : artifacts )
+                {
+                    printClassesFromArtifactResult( result );
+                }
+            }
+            else
+            {
+                ArtifactResult result = getArtifactResolver()
+                        .resolveArtifact( buildingRequest, toArtifactCoordinate( coordinate ) );
+
+                printClassesFromArtifactResult( result );
+            }
+        }
+        catch ( ArtifactResolverException | DependencyResolverException | IOException e )
+        {
+            throw new MojoExecutionException( "Couldn't download artifact: " + e.getMessage(), e );
+        }
+    }
+
+    private void printClassesFromArtifactResult( ArtifactResult result )
+            throws IOException
+    {
+        JarFile jarFile = new JarFile( result.getArtifact().getFile() );
+        Enumeration entries = jarFile.entries();
+
+        while ( entries.hasMoreElements() )
+        {
+            JarEntry entry = (JarEntry) entries.nextElement();
+            String name = entry.getName();
+
+            // filter out files that do not end in .class
+            if ( !name.endsWith( ".class" ) )
+            {
+                continue;
+            }
+
+            // remove .class from the end and change format to use periods instead of forward slashes
+            name = name.substring( 0, name.length() - 6 ).replace( '/', '.' );
```

Comment:
try to avoid reusing local variables. Here you have two things: an entry name and a class name, so two different variables make this clearer

Reply:
fixed